scanner.go

```diff
@@ -42,6 +42,26 @@ type Field struct {
 	IsEmbeded bool
 }
 
+type StructInfo struct {
+	Fields []Field
+}
+
+// GetStructInfo will return (and cache) informatoin about the given struct.
+//
+// `targetStruct` should either be a pointer to a struct type, or a
+// reflect.Type object of the structure in question
+func GetStructInfo(targetStruct interface{}) (si StructInfo, err error) {
```

Comment:
One question, why did you prefer to have a `StructInfo` struct instead of just returning a `[]Field`?

Reply:
Easier to add other things in the future - but don't feel strongly about this at all

Reply:
Yeah me neither, I think we can keep it as is, looks good.